src/event/cmap.rs

```diff
@@ -144,6 +156,15 @@ pub struct ConnectionClosedEvent {
 
     /// The reason that the connection was closed.
     pub reason: ConnectionClosedReason,
+
+    /// If the `reason` connection checkout failed was `Error`,the associated
+    /// error is contained here. This is attached so we can include it in log messages;
+    /// in future work we may add this to public API on the event itself. TODO: add
+    /// ticket link here.
+    #[cfg(feature = "tracing-unstable")]
+    #[serde(skip)]
+    #[derivative(PartialEq = "ignore")]
```

Comment:
we rely on this `PartialEq` implementation in the CMAP tests. but since `Error` does not implement `PartialEq` right now and this field is currently just here as an implementation detail for tracing it seemed like we could leave it out for now when comparing events in tests. 

that said, looking more closely, I'm not sure the `PartialEq` implementation is even needed. When I remove it from this type, the only thing that breaks is that the `CmapEvent` enum can't derive `PartialEq`. And if I remove the derived impl from that, the only thing that breaks is this line of code:

https://github.com/mongodb/mongo-rust-driver/blob/6e3d43d824facf8b051e3593d32f0ab378d095ef/src/cmap/test/mod.rs#L224

and IIUC all that line is doing is checking that the length of `subscriber.all(filter)` is 0??? so maybe we should just remove the `PartialEq` stuff altogether.

Reply:
Our `Serialize` / `Deserialize` impls are mostly there for test convenience, but I think something like `PartialEq` _might_ be functionality that a user could rely on, even if inadvertently like we are via a `Derive`. For that reason, I'd be hesitant to remove any of those implementations until 3.0, even if they're largely unnecessary.

Reply:
oh yeah, that's a good point. I wasn't thinking about this being a public type. I agree it seems best to keep it around then. filed RUST-1538 about considering a removal in 3.0